sdk/include/futex.h

```diff
@@ -67,10 +67,14 @@ __always_inline static int futex_wait(const uint32_t *address,
  * Wakes up to `count` threads that are sleeping with `futex[_timed]_wait` on
  * `address`.
  *
- * The `address` argument must permit storing four bytes of data after the
- * address. This call does not store to the address but requiring store
- * permission prevents a thread from waking up a futex that it cannot possibly
- * have moved to a different state.
+ * The `address` argument must have a length of at least four after the address
```

Comment:
May be worth more fully enumerating what `check_pointer` is going to ask for: valid, unsealed, and length >= 4.